sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations_base.py

```diff
@@ -654,44 +654,67 @@ def _populate_arm_parameters(self, workspace: Workspace, **kwargs: Dict) -> Tupl
             )
-            _set_val(
-                param["offline_store_connection_name"],
-                workspace._feature_store_settings.offline_store_connection_name
-                if workspace._feature_store_settings.offline_store_connection_name
-                else "",
-            )
-            _set_val(
-                param["online_store_connection_name"],
-                workspace._feature_store_settings.online_store_connection_name
-                if workspace._feature_store_settings.online_store_connection_name
-                else "",
-            )
+            if workspace._feature_store_settings.offline_store_connection_name:
+                _set_val(
+                    param["offline_store_connection_name"],
+                    workspace._feature_store_settings.offline_store_connection_name,
+                )
+            if workspace._feature_store_settings.online_store_connection_name:
+                _set_val(
+                    param["online_store_connection_name"],
+                    workspace._feature_store_settings.online_store_connection_name,
+                )
 
         if workspace._kind and workspace._kind.lower() == "featurestore":
             materialization_identity = kwargs.get("materialization_identity", None)
             offline_store_target = kwargs.get("offline_store_target", None)
             online_store_target = kwargs.get("online_store_target", None)
 
-            from azure.ai.ml._utils._arm_id_utils import AzureResourceId
+            from azure.ai.ml._utils._arm_id_utils import AzureResourceId, AzureStorageContainerResourceId
 
-            if offline_store_target is not None:
-                arm_id = AzureResourceId(offline_store_target)
-                _set_val(param["offline_store_target"], offline_store_target)
+            if offline_store_target:
+                arm_id = AzureStorageContainerResourceId(offline_store_target)
+                _set_val(param["offlineStoreStorageAccountOption"], "existing")
+                _set_val(param["offline_store_container_name"], arm_id.container)
+                _set_val(param["offline_store_storage_account_name"], arm_id.storage_account)
                 _set_val(param["offline_store_resource_group_name"], arm_id.resource_group_name)
                 _set_val(param["offline_store_subscription_id"], arm_id.subscription_id)
-            if online_store_target is not None:
+            else:
+                _set_val(param["offlineStoreStorageAccountOption"], "new")
+                _set_val(
+                    param["offline_store_container_name"],
+                    _generate_storage_container(workspace.name, resources_being_deployed),
+                )
+                if not workspace.storage_account:
+                    _set_val(param["offline_store_storage_account_name"], param["storageAccountName"]["value"])
```

Comment:
this is case the offline store shares same storage with default blob storage. How to enforce this?

Reply:
what do you mean by how to enforce?